source/agent/conference/roomController.js

```diff
@@ -1590,9 +1597,20 @@ module.exports.create = function (spec, on_init_ok, on_init_failed) {
                 }
             };
 
-            var spread2LocalNode = function (audioStream, videoStream, on_spread_ok, on_spread_error) {
-                log.debug('spread2LocalNode, subscriber:', terminal_id, 'audioStream:', audioStream, 'videoStream:', videoStream);
-                if (terminals[terminal_id] && (audioStream !== undefined || videoStream !== undefined)) {
+            var spread2LocalNode = function (audioStream, videoStream, dataStream, on_spread_ok, on_spread_error) {
+                log.debug('spread2LocalNode, subscriber:', terminal_id, 'audioStream:', audioStream, 'videoStream:', videoStream, ', dataStream: ', dataStream);
+                if (terminals[terminal_id] && dataStream) {
+                    const target_node = terminals[terminal_id].locality.node;
+                    const target_node_type = terminals[terminal_id].type;
+                    spreadStream(dataStream, target_node, target_node_type, () => {
+                        if (streams[dataStream] && terminals[terminal_id]) {
+                            on_spread_ok();
+                        } else {
+                            shrinkStream(dataStream, target_node);
+                            on_spread_error('Terminal or stream early left.');
+                        }
+                    }, on_error);
+                } else if (terminals[terminal_id] && (audioStream !== undefined || videoStream !== undefined || dataStream !== undefined)) {
```

Comment:
If `terminals[terminal_id]` is true and (dataStream !== undefined), is this condition same as the line 1602?

Reply:
It could be removed. It is there because firstly I wanted to reuse line 1613-1647, but later I found that there is special logical for audio and video. Data stream seems to be simpler.

Reply:
It has been removed in the latest commit.